src/field/Types.hh

```diff
@@ -37,4 +39,25 @@ struct StepperResult
 };
 
 //---------------------------------------------------------------------------//
+/*!
+ * The result of moving up to a certain distance along a step.
+ */
+struct DriverResult
+{
+    OdeState  state; //!< Post-step state
+    real_type step;  //!< Actual curved step
+};
+
+//---------------------------------------------------------------------------//
+// FUNCTIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Perform y <- ax + y for OdeState.
+ */
+inline CELER_FUNCTION void axpy(real_type a, const OdeState& x, OdeState* y)
```

Comment:
Should this be moved to FieldUtils or at least the axpy definition inside FieldUtils be removed.

Reply:
Whoops, thanks for the catch. The FieldUtils should definitely not have had the axpy. I defined `axpy` inside `Types` so that ADL works with OdeState when calling `axpy` on its own. In ORANGE I've defined small helper functions alongside the tiny structs defined in Types so I think it's OK there.